common/gcp.go

```diff
@@ -0,0 +1,86 @@
+package common
+
+import (
+	"fmt"
+	"net/http"
+
+	"golang.org/x/oauth2"
+	"google.golang.org/api/impersonate"
+)
+
+func (c *DatabricksClient) getGoogleOIDCSource() (ts oauth2.TokenSource, err error) {
+	// source for generateIdToken
+	ts, err = impersonate.IDTokenSource(c.InitContext, impersonate.IDTokenConfig{
+		Audience:        c.Host,
+		TargetPrincipal: c.GoogleServiceAccount,
+		IncludeEmail:    true,
+	}, c.googleAuthOptions...)
+	if err != nil {
+		err = fmt.Errorf("could not obtain OIDC token. %w Running 'gcloud auth application-default login' may help", err)
+		return
+	}
+	// TODO: verify that refreshers work...
+	ts = oauth2.ReuseTokenSource(nil, ts)
+	return
```

Comment:
return ts, nil ?

Reply:
I understand you are using named return arguments, but personally I think saying `return ts,nil` makes the code nicer to read

Reply:
On shorter methods we use named returns 🤷🏻‍♂️